middleware/auth.js

```diff
@@ -0,0 +1,61 @@
+export default async function (context) {
+  if (!context.from) {
+    if (context.route.query.code) {
+      context.app.$cookies.set('auth-token', context.route.query.code, {
+        secure: true,
+        sameSite: 'Strict',
+        httpOnly: true,
+      })
+
+      return context.redirect(context.route.path)
+    }
+
+    if (context.app.$cookies.get('auth-token')) {
+      const cookie = context.app.$cookies.get('auth-token')
+
+      await context.store.dispatch('auth/fetchUser', { token: cookie })
+    }
+  }
+
+  if (routeOption(context.route, 'auth', false)) return
+
+  if (!getMatchedComponents(context.route, []).length) {
+    return
+  }
+
+  if (!context.$auth.user) {
+    return context.redirect(
+      `https://api.modrinth.com/api/v1/auth/init?url=https://modrinth.com${context.route.fullPath}`
+    )
+  }
+}
+
+function routeOption(route, key, value) {
+  return route.matched.some((m) => {
+    if (process.client) {
+      // Client
+      return Object.values(m.components).some(
+        (component) => component.options && component.options[key] === value
+      )
+    } else {
+      // SSR
+      return Object.values(m.components).some((component) =>
+        Object.values(component._Ctor).some(
+          (ctor) => ctor.options && ctor.options[key] === value
+        )
+      )
+    }
+  })
+}
+
+function getMatchedComponents(route, matches) {
+  return [].concat(
+    ...[],
+    ...route.matched.map(function (m, index) {
```

Comment:
Why not arrow functions? ie `(param) => result`

Reply:
Could you suggest what it should look like?